Scripts/Runtime/Entities/TaskDriver/TaskSet/TaskData/DataStream/Cancellation/CancelBehaviour.cs

```diff
@@ -0,0 +1,39 @@
+namespace Anvil.Unity.DOTS.Entities.Tasks
+{
+    public enum CancelBehaviour
+    {
+        /// <summary>
+        /// When a cancellation request is received, the data is removed from the
+        /// <see cref="IAbstractDataStream"/> and is considered cancelled during the
+        /// consolidation phase.
+        /// </summary>
+        /// <remarks>
+        /// This is the most common case and should be used for data that does not need
+        /// to gracefully unwind when cancelled. 
+        /// </remarks>
+        Default,
+        /// <summary>
+        /// When a cancellation request is received, the data remains in the
+        /// <see cref="IAbstractDataStream"/> and cannot be cancelled. The data will
+        /// survive one iteration of the consolidation phase.
+        /// </summary>
+        /// <remarks>
+        /// A rare case but used in situations where custom cancel result data is needed.
+        /// For example, a Timer that when cancelled wants to communicate how much time was left
+        /// on the Timer. 
+        /// </remarks>
+        None,
+        /// <summary>
+        /// When a cancellation request is received, the data is removed from the
+        /// <see cref="IAbstractDataStream"/> and moved to a hidden internal
+        /// <see cref="IAbstractDataStream"/> during the consolidation phase.
+        /// </summary>
+        /// <remarks>
+        /// Instead of being deleted like in <see cref="Default"/>
+        /// it is copied to a hidden internal <see cref="IAbstractDataStream"/> that
+        /// can be used to explicitly cancel. This allows for custom jobs to run on
+        /// cancelled data to gracefully unwind.
+        /// </remarks>
+        Explicit
```

Comment:
How about this from the user level:

- Rename to `CancelRequestBehaviour` - This how it will behave to a Cancel Request.

The enums are now:

- Delete (previously Default) - This will just delete the element when it gets a cancel request
- Ignore (previously None) - This will ignore the cancel request and do nothing
- Unwind (previously Explicit) - This will launch the custom job to gracefully unwind the element.

I like the above and have implemented but feel free to comment further.

Reply:
These are good but I'm confused about Ignore. Maybe the example in the code is throwing me but if a timer needed to return a result why wouldn't it use `Unwind`?

Reply:
Ahhh i see the confusion. Yeah that's not a great comment if you don't live inside my head. Reworded. 

Also #146 will make it so that the end-developer will pretty much always only end up using `Delete` and `Unwind`.

Reply:
OK, I like the names and the improved comments are helpful. I've made some minor wording suggestions.
(for future readers this type was renamed to `CancelRequestBehaviour`)